LICENSE

```diff
@@ -1,17 +0,0 @@
-Permission is hereby granted, free of charge, to any person obtaining a copy
-of this software and associated documentation files (the "Software"), to
-deal in the Software without restriction, including without limitation the
-rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
-sell copies of the Software, and to permit persons to whom the Software is
-furnished to do so, subject to the following conditions:
-
-The above copyright notice and this permission notice shall be included in
-all copies or substantial portions of the Software.
```

Comment:
Doesn't removal violate this requirement?

Reply:
SPDX is a reference to a license in an external DB, not to a notice within a license. And the requirement is to include this notice in all copies of the project.

IANAL, but I'm like 99% sure downstreams would not be able to distribute said software if it doesn't have a license file.

It's probably a good idea to ask @hroncok @befeleme @mgorny if the respective distros have policies that would cause problems.

Also, GitHub will probably stop being able to detect said licenses. It uses https://licensee.github.io/licensee/ to perform detection. You can run it as a CLI tool in a container to see what it'd return.

Reply:
IANAL but the way I understand it, the notice works "downwards". Basically, if the project sources include one, then all redistributions of these sources must include one as well. So if the project sources no longer include the notice, then it simply means redistributions don't have to include it either.

That said, if the project has had third party contributions, then the situation *might* be different. In particular, since all contributions were made under the license in question and under the assumption that such a notice is present, then the notice effectively holds an obligation for the primary author from other authors. Therefore, unilaterally removing the notice without the agreement from other contributors could be perceived as a license violation.

Reply:
TLDR; In Fedora, this wouldn't make us too happy, but there are ways forward.

Fedora's licensing guidelines strongly state the preference of including the license file into the distribution, if the license itself states it must be distributed with copies of the software. We must include the file, if present, and ask upstream developers to include the file if it isn't there. If that doesn't render the result, we can either include a text of that well known license (ideally after confirming with upstream this is the correct text), or decide not to package such project. I believe MIT is an example of a license that requires going through the process.
Source: https://docs.fedoraproject.org/en-US/packaging-guidelines/LicensingGuidelines/#_license_text

Reply:
Thank you very much @befeleme.

> tag expression should be an enumeration of all licenses found in the source code of the package

I understand the motivation, but I don't think this is realistic or maintainable in the case of projects that have vendored code, that is why I am reluctant in adding it.

I am happy to ship all the license files, so that the correct licensing information is discoverable, but deriving and maintaining a SPDX license expression (specially when the bundled files can vary on updates) is not something I will be working on.